### PR TITLE
Fix for issue #966 , Chorf fixes

### PR DESCRIPTION
**Pull request by floskan**
Fixes hero scopes being incorrectly set to lords army, which makes the traits useless, fixes recruitment of heroes from landmarks for chaos dwarfs, fixes landmark effects not working for chaos dwarfs, fixes certain ancilleries for chaos dwarfs not working and fixes two convoy traits not working

The ancillaries are : Infernal engineer, Bull centaur guardian, K'daai hellsmith and Warsmith. 
The landmarks that gets fixed are Ancient inner city which now applies all it's bonuses (Recruit Taurok hero all provinces, recruit rank and cost reduction) same with Infernal barracks but( Infernal castellan recruit available factionwide) and flayed tower(1 PO factionwide).

The convoys fixed are metallic resilience, war-machinist and hobgoblin favour not applying their traits to the army.

### CODE REVIEW

**Comment by mfvreeland** [comment]
Could I get a little more detail on this one for the patch notes? Which landmarks are affected, which ancillaries, which convoy traits, etc. The hero scopes part is clear enough.

**Comment by mfvreeland** [comment]
@floskan 

The amount of traits with obvious scope issues that made it into release is really incredible. 

Couple questions before merging:

1. Why did you need to recreate the `wh3_dlc23_effect_chd_lord_effect_speed_war_machines` effect? I see that the unit set is missing, but the effect already existed.
2. Why did you need to make a new unit set for Chorf war machines? There is one already.

**Comment by floskan** [comment]
> @floskan
> 
> The amount of traits with obvious scope issues that made it into release is really incredible.
> 
> Couple questions before merging:
> 
>     1. Why did you need to recreate the `wh3_dlc23_effect_chd_lord_effect_speed_war_machines` effect? I see that the unit set is missing, but the effect already existed.
> 
>     2. Why did you need to make a new unit set for Chorf war machines? There is one already.

1. Thanks for noticing, that doesn't need to be there.
2. There is two different convoy traits, one that buffs hellcannon and then there is "War-machinist" so the unit set was recreated but excluding the hellcannons.

File: db/character_skill_level_to_effects_junctions_tables/zzz_cbfm_chd_traits_fix.tsv
Comment (mfvreeland) [inline db/character_skill_level_to_effects_junctions_tables/zzz_cbfm_chd_traits_fix.tsv:1]: I don't think the line for `wh3_dlc23_skill_innate_chd_convoy_overseer_hobgoblin_favour` needed to be added in this table, as it's already there in exactly the same form, but it won't hurt anything leave it.

**Pull request merged**